src/unity/python/turicreate/visualization/plot.py

```diff
@@ -126,33 +139,61 @@ def save(self, filepath, include_data=True):
         >>> plt.save('vega_spec.json', False)
 
         """
-        spec = _json.loads(self.__proxy__.get('call_function', {'__function_name__': 'get_spec'}))
-
-        if(include_data):
-            data = _json.loads(self.__proxy__.get('call_function', {'__function_name__': 'get_data'}))["data_spec"]
-
-            for x in range(0, len(spec["vega_spec"]["data"])):
-                if(spec["vega_spec"]["data"][x]["name"] == "source_2"):
-                    spec["vega_spec"]["data"][x] = data
-                    break;
-
-        with open(filepath, 'w') as fp:
-            _json.dump(spec, fp)
+        if type(filepath) != str:
+            raise ValueError("filepath provided is not a string")
+
+        if filepath.endswith(".json"):
+            # save as vega json
+            spec = self._get_vega(include_data = include_data)
+            with open(filepath, 'w') as fp:
+                _json.dump(spec, fp)
+        elif filepath.endswith(".png") or filepath.endswith(".svg"):
+            # save as png/svg, but json first
+            spec = self._get_vega(include_data = True)
+            EXTENSION_START_INDEX = -3
+            extension = filepath[EXTENSION_START_INDEX:]
+            temp_file_tuple = _mkstemp()
+            temp_file_path = temp_file_tuple[1]
+            with open(temp_file_path, 'w') as fp:
+                _json.dump(spec, fp)
+            dirname = _os.path.dirname(__file__)
+            relative_path_to_vg2png_vg2svg = "../vg2" + extension
+            absolute_path_to_vg2png_vg2svg = _os.path.join(dirname, 
+                relative_path_to_vg2png_vg2svg)
+            (exitcode, stdout, stderr) = _run_cmdline("node " + 
+                absolute_path_to_vg2png_vg2svg + " " 
+                + temp_file_path + " " + filepath)
+            if exitcode == NODE_NOT_FOUND_ERROR_CODE:
+                # user doesn't have node installed
+                raise NameError("Node.JS not found. Please download " + 
```

Comment:
Should be `Node.js` (lowercase js).

Reply:
And we should explain that saving as png and jpg (specifically) requires Node.js - otherwise it's implied that you need it to use Turi Create in general (which is not the case).